From f4093c714d3bf1d249bf7873c8c7c8a7e388a446 Mon Sep 17 00:00:00 2001
From: Alban Bedel <alban.bedel@avionic-design.de>
Date: Mon, 10 Sep 2018 13:22:09 +0200
Subject: [PATCH 3/4] crypto-ccm: crypto_memneq() has been backported to 3.12
 and 3.10

Adjust the #if to account for the backports to 3.12 and 3.10.

Signed-off-by: Alban Bedel <alban.bedel@avionic-design.de>
---
 compat/crypto-ccm.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/compat/crypto-ccm.c b/compat/crypto-ccm.c
index 4128e5f..7a77310 100644
--- a/compat/crypto-ccm.c
+++ b/compat/crypto-ccm.c
@@ -21,7 +21,11 @@
 #include <linux/slab.h>
 #include <linux/version.h>
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(3,13,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,13,0) && \
+  !(LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,59) && \
+    LINUX_VERSION_CODE <= KERNEL_VERSION(3,12,255)) && \
+  !(LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,106) && \
+    LINUX_VERSION_CODE <= KERNEL_VERSION(3,10,255))
 /* consider properly backporting this? */
 static int crypto_memneq(const void *a, const void *b, size_t size)
 {
-- 
2.17.1

